### PR TITLE
Add changelog handling to the `inmanta module release` command.

### LINKED ISSUE #5202: Improve changelog handling of release command
Tasks:

- [x] The new changelog entry added to the changelog after running the `inmanta module release` command should have the following format `## vx.y.z - ?` where `x.y.z` is the first patch release ahead of the currently released version.
- [x] When running `inmanta module release`, the question mark in the header of the changelog entry should be replaced with the current date (for example 2022/12/29).
- [x] When the new section is added, put it right above the section for the previous version, not necessarily at the top of the file.

### PR DESCRIPTION
**Pull request by arnaudsjs**
# Description

Add changelog handling to the `inmanta module release` command.

closes #5202 

# Self Check:

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [x] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

**Comment by arnaudsjs** [comment]
@edvgui I added you as a reviewer to see if the high level behavior matches the needs within the solutions team.

**Review by FloLey — APPROVED** [review]

**Review by edvgui — APPROVED** [review]
Very cool!

File: src/inmanta/moduletool.py [hunk 1]
```diff
@@ -1224,6 +1206,176 @@ def release(
...
+    The expected format of the changelog is the following:
+
+    ```
+    # Changelog
+
+    ## v1.2.1 - ?
+
+    - Change3
+
+    ## v1.2.0 - 2022/12/19
```
[the remark this answers is not in the document]
Comment (arnaudsjs) [reply src/inmanta/moduletool.py:1222]: Bart suggested to use the changelog from the connect module as a reference for this implementation. There the `/`-based format is used and I just took it over. But I don't think this is a big deal to change to the `-`-based format. @bartv @edvgui Any opinion?

[the remark this answers is not in the document]
Comment (edvgui) [reply src/inmanta/moduletool.py:1222]: Following the standard sounds better to me indeed

[the remark this answers is not in the document]
Comment (arnaudsjs) [reply src/inmanta/moduletool.py]: Fixed.

**Pull request closed without merging**